### ISSUE TITLE
Nordictrack Elite 1000

### ISSUE THREAD

**Issue by Skyla157** [cagnulein/QZCompanionNordictrackTreadmill#78]
I'm getting a response on the treadmill but it's not changing speed/incline.

**Comment by cagnulein** [comment]
hi please post the screenshot of your ifit interface on the manual screen
as i indicated on facebook
thanks
midnight here so i will answer back tomorrow

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno sab 27 gen 2024 alle 23:54 Meagan Sanchez <
***@***.***> ha scritto:

**Comment by Skyla157** [comment]
![IMG_0510 Large](https://github.com/cagnulein/QZCompanionNordictrackTreadmill/assets/16845105/0752523a-3380-46cd-af17-9f1f2a1e9b5c)

**Comment by Skyla157** [comment]
Apologies for poor lighting. let me know if you need a better lit picture.

**Comment by cagnulein** [comment]
yes i need a picture from adb, i need the precise pixel measurements unfortunately

**Comment by Skyla157** [comment]
![IMG_0512 Large](https://github.com/cagnulein/QZCompanionNordictrackTreadmill/assets/16845105/4efd9bb1-6718-46a0-a454-f62048ea67ab)
This?

**Comment by cagnulein** [comment]
you have to run the script from the install package folder
so change the directory first on a prompt and then run it

i'm going to bed now, @victorypoint are you able to assist? otherwise i will answer you back tomorrow

**Comment by Skyla157** [comment]
![IMG_0516](https://github.com/cagnulein/QZCompanionNordictrackTreadmill/assets/16845105/a9e2cf20-cb9e-4e1a-a1db-2b74477bc1df)
Is this correct?

**Comment by victorypoint** [comment]
That looks good @Skyla157. You need to run adb-screenshot.bat so it will capture and save a screenshot of your treadmill screen to the same folder.

**Comment by Skyla157** [comment]
![Sat01-27-2024@15-41-10 16](https://github.com/cagnulein/QZCompanionNordictrackTreadmill/assets/16845105/089b8546-70b6-416e-9f3f-c4d7aa58f849)

**Comment by victorypoint** [comment]
Perfect @Skyla157. Thank-you. We'll get to work on this.

**Comment by cagnulein** [comment]
@Skyla157 patch done, download the new companion https://github.com/cagnulein/QZCompanionNordictrackTreadmill/releases/download/3.3.31/QZCompanionNordictrackTreadmill.apk

select the elite 1000 device and reboot it and try

maybe we have to slightly adjust the coordinates, but it should be more or less right

**Comment by Skyla157** [comment]
Thank you for working on this! I’m actually about to head to bed but I wanted to give it a try. I did the update via my Mac and after the treadmill rebooted I started a manual workout, the QZ app on my phone, and then a 5 min cool down walk on the peloton app on my iPad. My phone connected to both treadmill and peloton workout. But no response to changes. I then did a hard reset on the treadmill and restarted the apps. Now it changes the incline! But no speed. Please let me know if there’s something else I should be doing. I’m heading to bed. Will check in in about eight hours or so. Thank you again!On Jan 27, 2024, at 11:21 PM, Roberto Viola ***@***.***> wrote:﻿
@Skyla157 patch done, download the new companion https://github.com/cagnulein/QZCompanionNordictrackTreadmill/releases/download/3.3.31/QZCompanionNordictrackTreadmill.apk
select the elite 1000 device and reboot it and try
maybe we have to slightly adjust the coordinates, but it should be more or less right

**Comment by cagnulein** [comment]
great! for the speed you need to change the force speed setting in the treadmill setting in qz
before testing on a real test, be sure that qz is changing correctly the speed (just to put safety first!)

**Comment by Skyla157** [comment]
Ok, incline definitely works (a bit delayed but that’s understandable). I turned on Treadmill Speed Forcing. Still can’t control speed through app and no changes during call outs. When I tap the plus sign, the treadmill screen pulses slightly. When I press and hold for a few seconds the track view goes to full screen. Also, probably due to me tapping buttons, but about three minutes in, the treadmill screen showed a message that QZ unexpectedly quit. 🙁On Jan 28, 2024, at 12:35 AM, Roberto Viola ***@***.***> wrote:﻿
great! for the speed you need to change the force speed setting in the treadmill setting in qz
before testing on a real test, be sure that qz is changing correctly the speed (just to put safety first!)

**Comment by cagnulein** [comment]
so the issue is that qz can't control speed at all even with plus sign?
where the touch event is? are you able to see it?

for the crash, don't mind, the companion is restarting by itself

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 28 gen 2024 alle 18:40 Meagan Sanchez <
***@***.***> ha scritto:

**Comment by Skyla157** [comment]
Yeah, the QZ app on my phone can’t control the speed. When I tap the plus
or minus, I see a response on the treadmill screen in that I can see the
display pulse slightly outwards from the center of the screen. And a long
press brings up the track screen as a full screen. But it doesn’t affect
speed.

**Comment by cagnulein** [comment]
ok so you are saying that the X coordinate is wrong. do i get it right?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 28 gen 2024 alle 18:51 Meagan Sanchez <
***@***.***> ha scritto:

**Comment by Skyla157** [comment]
Yes, that would be my assumption. Let me know if there’s anything I can test on my end to help!

On Jan 28, 2024, at 9:52 AM, Roberto Viola ***@***.***> wrote:


ok so you are saying that the X coordinate is wrong. do i get it right? 

Roberto Viola 
Software engineer and open source enthusiast 
http://robertoviola.cloud 



Il giorno dom 28 gen 2024 alle 18:51 Meagan Sanchez < 
***@***.***> ha scritto:

**Comment by cagnulein** [comment]
ok tomorrow i will check them again! thanks for the feedback!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 28 gen 2024 alle 18:55 Meagan Sanchez <
***@***.***> ha scritto:

> Yes, that would be my assumption. Let me know if there’s anything I can
> test on my end to help!
>

**Comment by cagnulein** [comment]
@Skyla157 already fixed! i'm providing a new apk!

**Comment by cagnulein** [comment]
@Skyla157  here https://github.com/cagnulein/QZCompanionNordictrackTreadmill/releases/download/3.3.32/QZCompanionNordictrackTreadmill.apk

**Comment by victorypoint** [comment]
@cagnulein, let me know if I can be of any help here.

**Comment by cagnulein** [comment]
i guess it's ok now @victorypoint thanks!

**Comment by Skyla157** [comment]
Nope, still not working. And now incline isn’t working through the QZ app either. Besides updating the treadmill app and rebooting, is there anything else I should do?On Jan 28, 2024, at 12:30 PM, Roberto Viola ***@***.***> wrote:﻿
i guess it's ok now @victorypoint thanks!

**Comment by cagnulein** [comment]
yes I guess you have to reboot it as you did yesterday, the only thing that I changed is the X coordinate for the speed @Skyla157

**Comment by Skyla157** [comment]
OK I’ll try a full reboot of all the devices. My husband just jumped on for his long run so I’ll give it a go in a few hours. Thank you again. I’m so excited to get this working!!On Jan 28, 2024, at 12:37 PM, Roberto Viola ***@***.***> wrote:﻿
yes I guess you have to reboot it as you did yesterday, the only thing that I changed is the X coordinate for the speed @Skyla157

**Comment by Skyla157** [comment]
Ok, I rebooted all my devices. The iPhone app shows the Peloton class title and the treadmill's speed and incline, (so everything connects properly) but I can't adjust either speed nor incline through the app and it doesn't auto change either.

**Comment by victorypoint** [comment]
@Skyla157, can you confirm with us that privileged mode and USB debugging is enabled on your treadmill, and also the on-screed sliders have to be visible. Let us know.

**Comment by Skyla157** [comment]
Yup! I entered privileged mode at the start of all this. USB debugging is still toggled on (I just checked). The sliders appear when I start a manual workout. The interesting thing though is that after I start the manual workout, I'll open up the QZ app and the sliders grey out. Like they aren't active. I then tap the treadmill screen and a quick flash appears (like the pause menu and then quickly dismissed?) and the sliders go back to black. So there's some input it's getting from the app. I can't replicate it though with my own taps.

**Comment by victorypoint** [comment]
@Skyla157, can you confirm that you're incline slider values go from zero to 12%, and speed values go from 1 to 12?

**Comment by Skyla157** [comment]
Yes, that is correct.

**Comment by victorypoint** [comment]
Okay good. @Skyla157, please download the attached file, unzip it into your QZ Companion folder. Set treadmill to manual workout with incline at 0% and then execute the unzipped script. It should move your incline slider up to 2%. Let us know.

[adb-swipe.zip](https://github.com/cagnulein/QZCompanionNordictrackTreadmill/files/14078533/adb-swipe.zip)

**Comment by Skyla157** [comment]
OK, it moved to 1% incline.

**Comment by victorypoint** [comment]
@Skyla157 good! I need to see the bottom square in the screenshot. Can you move the incline slider higher to show the entire bottom box and then upload a new screenshot? Sorry to be a hassle.

**Comment by Skyla157** [comment]
![Sun01-28-2024@17-06-50 65](https://github.com/cagnulein/QZCompanionNordictrackTreadmill/assets/16845105/bc7a6d9c-4300-4652-a7bd-9ae6320fbb4e)
Like this? Note the grayed out toggles. Not sure that happens usually - I don't typically show the sliders.

Don't worry about any hassle. This'll be totally worth it if we can get it working!

**Comment by victorypoint** [comment]
@Skyla157 perfect thank-you! Please download the attached and run again to confirm where the incline jumps to. The 1st jump should to to 1% and the second should be close to 12%.

[adb-swipe.zip](https://github.com/cagnulein/QZCompanionNordictrackTreadmill/files/14078746/adb-swipe.zip)

**Comment by Skyla157** [comment]
Yup! starts at 0, goes to 1%, then a second or two later, jumps to 12%

**Comment by victorypoint** [comment]
@Skyla157, okay awesome. Because your incline boxes go from 0 to 12 and skip 6, there is a ramp that we need to figure out in pixels. Are you comfortable with the command prompt in Windows, and navigate into the companion install folder and run commands from there?

**Comment by Skyla157** [comment]
sure!

**Comment by victorypoint** [comment]
Perfect! Ok, so in the install folder in command-line, confirm if the ADB connection is still alive from running the .bat file previously:

adb devices -l

If no good then:
adb connect [ip address of treadmill]

If good, set treadmill to manual workout and 0% incline, then let's proceed to:
adb shell input swipe 76 589 76 556 200

This goes from 0 to 1%. What this means is the y-pixel starting location is 589, the y-pixel ending location is 556. I'm simply subtracting 33 which is the y distance between boxes. I need you to try the adb swipes and map out the y-pixel values for each 0.5% change in incline. It's a bit tedious work unfortunately. Make sure to reset incline back to 0% before trying each swipe.

For 12% for example, I'm using:
adb shell input swipe 76 589 76 226 200

Hope this helps. Let us know how it goes.

**Comment by Skyla157** [comment]
It seems like it's about every 15 or so pixels. Here's what worked for me here.

545 - 1.5
530 - 2.0
510 - 2.5
490 - 3.0
470 - 3.5
450 - 4.0
435 - 4.5
420 - 5.0
405 - 5.5
390 - 6.0
375 - 6.5
355 - 7.0
340 - 7.5
325 - 8.0
310 - 8.5
295 - 9.0
270 - 9.5
255 - 10.0
240 - 10.5
225 - 11.0
210 - 11.5
195 - 12.0

**Comment by victorypoint** [comment]
@Skyla157 thank-you!!! Looking at lowest value of 589 for 0%, and highest value of 195 for 12%, I get:

589 - 195/24 = 16.4 pixels between 0.5% jumps. @cagnulein can you use these values in a lookup table or prefer a calculation?

**Comment by victorypoint** [comment]
Thank-you @Skyla157. I'm off to bed but amazing progress was made tonight!

**Comment by cagnulein** [comment]
hi guys thanks for your work! 
I updated the apk with the new coordinates https://github.com/cagnulein/QZCompanionNordictrackTreadmill/releases/download/3.3.33/QZCompanionNordictrackTreadmill.apk

There is a thing that I don't understand: I thought inclination was already working with the first version? Or it was not? Also what about the speed?

Also I don't know if there is an udp issue since for these test you were using adb instead of udp.
Anyway let's see if this version is better.
Let me know

**Comment by Skyla157** [comment]
Hey guys, apologies for the long delay. Been a really busy few weeks. 

However I still haven't been able to get this to work. Speed and Incline tiles show the correct numbers from my treadmill, updating a few seconds after I make adjustments on the treadmill. However, nothing happens when i try to change the numbers on the QZ app. (Except sometimes the Companion app crashes on the treadmill.) Is there anything else we can try? 

@cagnulein yes, the incline was working on that first update but hasn't worked since.

**Comment by cagnulein** [comment]
@Skyla157 Do you still see a touch reaction on the 
 screen of the treadmill when you are pressing the inclination plus button on qz?

**Comment by Skyla157** [comment]
No. Except the companion crashes sometimes

**Comment by cagnulein** [comment]
Adb shell input commands are still working?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 15 feb 2024 alle 02:32 Meagan Sanchez <
***@***.***> ha scritto:

**Comment by Skyla157** [comment]
Yes, USB debugging is still toggled on and I can run commands via terminal. The IP address has changed but I don't think that would make a difference?

**Comment by cagnulein** [comment]
@Skyla157 i received a notification but i don't see the message . did you remove it?

**Comment by Skyla157** [comment]
Yes, I initially said everything was still connected normally. USB debugging toggled on and I ran some commands via terminal. One of them was to update to the latest version of QZ companion. The only difference was that the IP address changed, but that shouldn't affect anything? After I hit send, decided to try another walk on peloton and it's working! Not sure what changed as I've restarted the treadmill multiple times over the last few weeks. Maybe something in the latest updates?
